Library/ViewModels/ReportProjectFormViewModel.swift

```diff
@@ -53,33 +54,40 @@ public final class ReportProjectFormViewModel: ReportProjectFormViewModelType,
       .assign(to: &$retrievedEmail)
 
     /// Submits report on saveTriggered
-    self.$saveTriggered
-      .filter { $0 } // only save if saveTriggered is true
+    self.saveTriggeredSubject
       .compactMap { [weak self] _ in
         self?.createFlaggingInput()
       }
-      .flatMap { $0 }
-      .sink { [weak self] _ in
-        // An error happens
-
-        self?.saveTriggered = false
-        self?.bannerMessage = MessageBannerViewViewModel((
-          type: .error,
-          message: Strings.Something_went_wrong_please_try_again()
-        ))
-        self?.saveButtonEnabled = true
-
-      } receiveValue: { [weak self] _ in
+      .flatMap { [weak self] createFlaggingInput in
+        createFlaggingInput.catch { _ in
+          // An API error happens.
+          // We need to catch this up here in flatMap, instead of in sink,
+          // because we don't want an API failure to cancel this pipeline.
+          // If the pipeline gets canceled, you can't re-submit after a failure.
```

Comment:
This was something interesting I noticed while testing. Any error in a pipeline cancels/finishes the entire pipeline.